bridges/snowbridge/primitives/inbound-queue/src/v2/converter.rs

```diff
@@ -353,14 +336,17 @@ where
 		}
 
 		// Add the XCM sent in the message to the end of the xcm instruction
-		instructions.extend(message.remote_xcm.0);
+		instructions.extend(message.remote_xcm.clone().0);
 		instructions.push(RefundSurplus);
```

Comment:
As a convention, the topic should always be the last instruction in the xcm message. Various pallets look for the topic in that location.

Suppose `remote_xcm` includes a topic. By adding `RefundSurplus` after the instructions in `remote_xcm`, the topic becomes hidden.

You'll want to extract the topic in `remote_xcm` and add it as the last item in `instruction`.

Reply:
Addressed in https://github.com/vgeddes/polkadot-sdk/commit/d64dc321b56533a29b38d411a6b9275f3d9155d4.